packages/docusaurus-plugin-content-docs/src/plugin-content-docs.d.ts

```diff
@@ -137,3 +137,32 @@ declare module '@theme/Seo' {
   const Seo: (props: Props) => JSX.Element;
   export default Seo;
 }
+
+declare module '@theme/hooks/useDocs' {
```

Comment:
All this is not in a very good state currently and I'd like to have content plugins expose more easily code/types/hooks to the theme.

I'd like to avoid such type duplication, as it's painful and error-prone to maintain.

What about creating an `indexClient.ts`, exporting types that we need there/code/hooks, and migrate away from `plugin-content-docs.d.ts` file? (ie also update `package.types` => `indexClient.d.ts`)

If it's too complicated, one possibility could be to export the existing types instead of redeclaring them?

```ts
declare module '@theme/hooks/useDocs' {
  export type * from "./theme/hooks/useDocs"
}
```

Reply:
I agree with your concerns about code duplication, but not sure I understood your first solution fully. Since `plugin-content-docs.d.ts` contains almost all `@theme/*` modules except the `@docusaurus/plugin-content-docs-types` module, all the theme interfaces will still be exported from the same file, maybe just renamed.

I also think the typing system is quite inconsistent across the content plugins. The pages plugin has a similar `src/plugin-content-pages.d.ts` file, but the blog has an `/index.d.ts` file. Still, the docs plugin is the only one exporting a theme hook, causing all these complications.

The second solution only works for the type exports, but most of the exports here are function exports. I don't know how to handle this elegantly either...

Reply:
Going to merge this for now and try to figure this out later ;)